src/server/debug/server/server.go

```diff
@@ -0,0 +1,65 @@
+package server
+
+import (
+	"fmt"
+	"runtime/pprof"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"github.com/pachyderm/pachyderm/src/client/debug"
+	"github.com/pachyderm/pachyderm/src/client/pkg/grpcutil"
+	"github.com/pachyderm/pachyderm/src/server/worker"
+)
+
+// NewDebugServer creates a new server that serves the debug api over GRPC
+func NewDebugServer(name string, etcdClient *etcd.Client, etcdPrefix string) debug.DebugServer {
+	return &debugServer{
+		name:       name,
+		etcdClient: etcdClient,
+		etcdPrefix: etcdPrefix,
+	}
+}
+
+type debugServer struct {
+	name       string
+	etcdClient *etcd.Client
+	etcdPrefix string
+}
+
+func (s *debugServer) Dump(request *debug.DumpRequest, server debug.Debug_DumpServer) error {
+	profile := pprof.Lookup("goroutine")
+	if profile == nil {
+		return fmt.Errorf("unable to find goroutine profile")
+	}
+	w := grpcutil.NewStreamingBytesWriter(server)
+	if s.name != "" {
+		if _, err := fmt.Fprintf(w, "== %s ==\n\n", s.name); err != nil {
+			return err
+		}
+	}
+	if err := profile.WriteTo(w, 2); err != nil {
+		return err
+	}
+	if !request.Recursed {
+		request.Recursed = true
+		cs, err := worker.Clients(server.Context(), "", s.etcdClient, s.etcdPrefix)
```

Comment:
Not sure if you want to ever provide the pipeline RC name here. That might be a useful flag for debugging.

Reply:
We could possibly make that an argument... I may punt on this to see if it's actually needed. I mostly want this command so I have a quick way to get users to give me a goro dump, and I can filter out the parts I want from there.

Reply:
If there's a way to filter the output based on pipeline, then yea I don't think we need it. I assumed there wasn't since you the pipeline name you emit in the output comes from this value.

Reply:
I add the pod name ahead of all the stacks I dump, so things are delimited. Definitely could be nicer but I think just having a way to dump this and then doing manual filtering after the fact will get us most of what we need for debugging.